File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
def solution(progresses, speeds):
    answer = []

    arr = [-((p - 100) // s) for p, s in zip(progresses, speeds)]

    arr= deque(arr)

    while arr :
        a = []
        p = arr.popleft()
        a.append(p)
        if arr :
            while arr and p >= arr[0]:
                p2 = arr.popleft()
                a.append(p2)
        answer.append(len(a))
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayDeque<Integer> que = new ArrayDeque<>();
        int n = progresses.length;
        int[] dataLeft = new int[n];
        
        for(int i=0; i<n; i++){
            dataLeft[i] = (int) Math.ceil((100.0 - progresses[i]) / speeds[i]);
        }
        int count = 0;
        int maxLeft = dataLeft[0];
        
        for(int i=0; i<n; i++){
            
            if(dataLeft[i] <= maxLeft){
                count++;    
            }else{
                que.add(count);
                count = 1;
                maxLeft = dataLeft[i];
            }
        }
        que.add(count);
        return que.stream().mapToInt(Integer::intValue).toArray();
    }
}